clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -719,26 +846,27 @@ NetworkClientDelegate.UnsentRequest buildRequestWithResponseHandling(final Abstr
 
         private void handleClientResponse(final ClientResponse response,
                                           final Throwable error,
-                                          final long currentTimeMs) {
+                                          final long requestCompletionTimeMs) {
             try {
                 if (error == null) {
                     onResponse(response);
                 } else {
                     log.debug("{} completed with error", requestDescription(), error);
-                    handleCoordinatorDisconnect(error, currentTimeMs);
-                    if (error instanceof RetriableException) {
-                        maybeRetry(currentTimeMs, error);
-                    } else {
-                        future().completeExceptionally(error);
-                    }
+                    onFailedAttempt(requestCompletionTimeMs);
+                    handleCoordinatorDisconnect(error, requestCompletionTimeMs);
+                    future().completeExceptionally(error);
                 }
             } catch (Throwable t) {
-                log.error("Unexpected error handling response for ", requestDescription(), t);
+                log.error("Unexpected error handling response for {}", requestDescription(), t);
                 future().completeExceptionally(t);
             }
         }
 
         abstract void onResponse(final ClientResponse response);
+
+        abstract boolean retryTimeoutExpired(long currentTimeMs);
```

Comment:
Considered moving `expirationTimeout` to this super-class and inlining this? After all, expiration timeout is functionality that really belongs with the retry logic